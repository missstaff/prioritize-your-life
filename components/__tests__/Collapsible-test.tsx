import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import { Collapsible } from "../Collapsible";

describe("Collapsible", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(<Collapsible title="title">Content</Collapsible>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("toggles open and close when header is pressed", () => {
    const { getByTestId, queryByTestId } = render(
      <Collapsible title="title">Content</Collapsible>
    );

    expect(queryByTestId("collapsible-content")).toBeNull();

    fireEvent.press(getByTestId("collapsible-header"));

    expect(queryByTestId("collapsible-content")).not.toBeNull();

    fireEvent.press(getByTestId("collapsible-header"));

    expect(queryByTestId("collapsible-content")).toBeNull();
  });
});
