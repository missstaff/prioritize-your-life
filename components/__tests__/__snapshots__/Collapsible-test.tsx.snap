// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#151718",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 6,
        "opacity": 1,
      }
    }
    testID="collapsible-header"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#ECEDEE",
          },
          undefined,
          undefined,
          {
            "fontSize": 34.285714285714285,
            "fontWeight": "600",
            "lineHeight": 51.42857142857143,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;
